node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1232,7 +1265,14 @@ async fn fetch_blocks_and_state_since_genesis(ctx: &ChainSyncContext<'_>) -> Res
     let latest_height_requested: Arc<AtomicU64> = Arc::new(AtomicU64::new(0));
 
     let mut workers: FuturesUnordered<_> = (0..ctx.config.max_parallel_block_fetches())
-        .map(|worker_id| fetch_block_worker(worker_id, latest_height_requested.clone(), ctx))
+        .map(|worker_id| {
+            fetch_block_worker(
+                worker_id,
+                latest_height_requested.clone(),
+                validator_weights.clone(),
```

Comment:
Aren't those the wrong weights for this block — the ones from the latest switch block we downloaded?

Reply:
Should we just use `get_switch_block_by_era_id` instead? Or is that too expensive to do in every single worker job?

Reply:
> Or is that too expensive to do in every single worker job?

I had suggested caching the _per-era validators_ in the context but @Fraser999 and @marc-casperlabs thought we should try with the simple approach first and optimize later.

Reply:
The code is now using the `get_switch_block_header_at_era_id_from_storage`.